### DISCUSSION TITLE
Need Help for MCP23017

### DISCUSSION THREAD

**Discussion by andi-riese** [Pi4J/pi4j#416] [General]
Hi,

I use several MCP23017 on I2C bus 1 in my home automation system, which is currently still based on pi4j-v1. I would now like to switch to pi4j-v2.
Unfortunately, this is probably more complicated than I thought. I can't make sense of the example 
.../pi4j-examples/pi4j-example-devices_not_used/src/main/java/com/pi4j/devices/mcp23017
Actually, I just need to know how to read or write a pin A0-A7 or B0-B7, or the registers in the MCP23017. 
Can anyone help me?
Thanks in advance.
Andi

**Comment by taartspi** [comment]
https://github.com/Pi4J/pi4j-example-devices    This is the link to the examples. You looked at the mcp23017 example and this uses every register in that chip. And thar chip has a great deal of function.  I do have a simple use of that chip where i only wrote code for specific function. Maybe this is a starting point for you.   
MCP23017_LCD1602A
